include/libopencm3/stm32/l4/usart.h

```diff
@@ -0,0 +1,519 @@
+
+/* Interrupt flag clear register (USARTx_ICR) */
+#define USART_ICR(usart_base)		MMIO32((usart_base) + 0x20)
+#define USART1_ICR			USART_ICR(USART1_BASE)
+#define USART2_ICR			USART_ICR(USART2_BASE)
+#define USART3_ICR			USART_ICR(USART3_BASE)
+#define UART4_ICR			USART_ICR(UART4_BASE)
+#define UART5_ICR			USART_ICR(UART5_BASE)
+#define LPUART1_ICR			USART_CR1(LPUART1_BASE)
+
+/* Receive data register (USARTx_RDR) */
+#define USART_RDR(usart_base)		MMIO32((usart_base) + 0x24)
+#define USART1_RDR			USART_RDR(USART1_BASE)
+#define USART2_RDR			USART_RDR(USART2_BASE)
+#define USART3_RDR			USART_RDR(USART3_BASE)
+#define UART4_RDR			USART_RDR(UART4_BASE)
+#define UART5_RDR			USART_RDR(UART5_BASE)
+#define LPUART1_RDR			USART_CR1(LPUART1_BASE)
+
+/* Transmit data register (USARTx_TDR) */
+#define USART_TDR(usart_base)		MMIO32((usart_base) + 0x28)
+#define USART1_TDR			USART_TDR(USART1_BASE)
+#define USART2_TDR			USART_TDR(USART2_BASE)
+#define USART3_TDR			USART_TDR(USART3_BASE)
+#define UART4_TDR			USART_TDR(UART4_BASE)
+#define UART5_TDR			USART_TDR(UART5_BASE)
+#define LPUART1_TDR			USART_CR1(LPUART1_BASE)
+
+
+/* --- USART_CR1 values ----------------------------------------------------- */
+/* --- (dsparse v1.1 at Thu Jun 22 12:28:02 2017) --- */
+
+/* M1: Word length */
+#define USART_CR1_M1			(1 << 28)
+
+/* EOBIE: End of Block interrupt enable */
+#define USART_CR1_EOBIE			(1 << 27)
+
+/* RTOIE: Receiver timeout interrupt enable */
+#define USART_CR1_RTOIE			(1 << 26)
+
+/* DEAT[4:0]: Driver Enable assertion time */
+#define USART_CR1_DEAT_MASK		(0x1F << 21)
+
+/* DEDT[4:0]: Driver Enable de-assertion time */
+#define USART_CR1_DEDT_MASK		(0x1F << 16)
+
+/* OVER8: Oversampling mode */
+#define USART_CR1_OVER8			(1 << 15)
+
+/* CMIE: Character match interrupt enable */
+#define USART_CR1_CMIE			(1 << 14)
+
+/* MME: Mute mode enable */
+#define USART_CR1_MME			(1 << 13)
+
+/* M0: Word length */
+#define USART_CR1_M			(1 << 12)
+
+/* WAKE: Receiver wakeup method */
+#define USART_CR1_WAKE			(1 << 11)
+
+/* PCE: Parity control enable */
+#define USART_CR1_PCE			(1 << 10)
+
+/* PS: Parity selection */
+#define USART_CR1_PS			(1 << 9)
+
+/* PEIE: PE interrupt enable */
+#define USART_CR1_PEIE			(1 << 8)
+
+/* TXEIE: interrupt enable */
+#define USART_CR1_TXEIE			(1 << 7)
+
+/* TCIE: Transmission complete interrupt enable */
+#define USART_CR1_TCIE			(1 << 6)
+
+/* RXNEIE: RXNE interrupt enable */
+#define USART_CR1_RXNEIE		(1 << 5)
+
+/* IDLEIE: IDLE interrupt enable */
+#define USART_CR1_IDLEIE		(1 << 4)
+
+/* TE: Transmitter enable */
+#define USART_CR1_TE			(1 << 3)
+
+/* RE: Receiver enable */
+#define USART_CR1_RE			(1 << 2)
+
+/* UESM: USART enable in Stop mode */
+#define USART_CR1_UESM			(1 << 1)
+
+/* UE: USART enable */
+#define USART_CR1_UE			(1 << 0)
+
+
+/* --- USART_CR2 values ----------------------------------------------------- */
+/* --- (dsparse v1.1 at Thu Jun 22 12:28:02 2017) --- */
```

Comment:
drop all of these please, they're not interesting here.

If you add a second * to your comments on the defines, they'll autogenerate doxygen properly.